File: app/src/main/kotlin/de/anubi1000/turnierverwaltung/data/participant/Participant.kt
package de.anubi1000.turnierverwaltung.data.participant

import de.anubi1000.turnierverwaltung.database.model.ParticipantModel
import org.mongodb.kbson.ObjectId

data class Participant(
    val id: ObjectId = ObjectId(),
    val tournamentId: ObjectId = ObjectId(),
    val name: String = "",
    val verein: String = ""
)

fun ParticipantModel.toParticipant(): Participant = Participant(
    id = id,
    name = name,
    verein = verein
)

File: Server/Database/Model/Participant.cs
namespace Turnierverwaltung.Server.Database.Model;

public sealed class Participant
{
    // Common properties
    public int Id { get; init; }

    public string Name { get; set; } = "";

    public int StartNumber { get; set; }
    public Gender Gender { get; set; } = Gender.Male;

    // Tournament
    public int TournamentId { get; init; }
    public Tournament Tournament { get; init; } = null!;

    // Club
    public int ClubId { get; set; }
    public Club Club { get; init; } = null!;

    // Results
    public ICollection<ParticipantResult> Results { get; init; } = new List<ParticipantResult>();
}
